streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java

```diff
@@ -145,13 +152,47 @@ public String getIngressModeDefault() {
 
   @Override
   public ResponseResult checkDocker(DockerConfig dockerConfig) {
-    // TODO check
     ResponseResult result = new ResponseResult();
-    result.setStatus(200);
-    result.setMsg("success");
+
+    CloseableHttpClient httpclient = HttpClients.createDefault();
+
+    // Transfer a JSON object with the login information to the server
+    HttpPost loginRequest = getHttpPost(dockerConfig);
+
+    try {
+      CloseableHttpResponse loginResponse = httpclient.execute(loginRequest);
+
+      int statusCode = loginResponse.getStatusLine().getStatusCode();
+
+      if (statusCode == 200) {
+        result.setStatus(200);
+        result.setMsg("successful");
+      } else {
+        result.setStatus(statusCode);
+        result.setMsg("login to server failed please check");
+      }
+    } catch (IOException e) {
+      result.setStatus(500);
+      result.setMsg("request failed, please check: " + e.getMessage());
+    }
+
     return result;
   }
 
+  private static HttpPost getHttpPost(final DockerConfig dockerConfig) {
+    HttpPost loginRequest = new HttpPost(dockerConfig.getAddress());
+    loginRequest.setHeader("Content-Type", "application/json");
+    String userLoginParam =
```

Comment:
The username and password set in this way are invalid. We need to login for verification through the docker-java API. see this [PR](https://github.com/apache/incubator-streampark/pull/3593)